Sources/StreamChatUI/Utils/VideoLoading.swift

```diff
@@ -6,21 +6,32 @@ import AVKit
 import StreamChat
 import UIKit
 
-/// A protocol the video preview uploader implementation must conform to.
-public protocol VideoPreviewLoader: AnyObject {
+/// A protocol the video loader must conform to.
+public protocol VideoLoading: AnyObject {
     /// Loads a preview for the video at given URL.
     /// - Parameters:
     ///   - url: A video URL.
     ///   - completion: A completion that is called when a preview is loaded. Must be invoked on main queue.
     func loadPreviewForVideo(at url: URL, completion: @escaping (Result<UIImage, Error>) -> Void)
+    
+    /// Returns a video asset with the given URL.
+    ///
+    /// - Returns: The video asset.
+    func videoAsset(at url: URL) -> AVURLAsset
+}
+
+public extension VideoLoading {
+    func videoAsset(at url: URL) -> AVURLAsset {
+        .init(url: url)
+    }
 }
 
-/// The `VideoPreviewLoader` implemenation used by default.
-final class DefaultVideoPreviewLoader: VideoPreviewLoader {
+/// The `VideoLoading` implementation used by default.
+open class StreamVideoLoader: VideoLoading {
     private let cache: Cache<URL, UIImage>
     
-    init(countLimit: Int = 50) {
-        cache = .init(countLimit: countLimit)
+    public init(cachedVideoPreviewsCountLimit: Int = 50) {
```

Comment:
I think the required init should be `public init(cache: Cache<URL, UIImage> = .init(countLimit: 50)`. This makes it more testable and clear what is the dependencies of `StreamVideoLoader`.

Has a convenience init we can have the `public init(cachedVideoPreviewsCountLimit: Int = 50)` still 👍

Reply:
I agree with both args 👍 But this would require opening `Cache` type, right? 🤔 And since `Cache` is very basic it'd also be nice to hide it behind a protocol and let customers provide their own caching. But because of `Cache` being generic it will involve some extra work. And also increase the public-facing APIs with smth not directly related to chats. Not sure if it's the right move.

Reply:
Ah I see, nevermind then, at least for now, we can leave it like this 👍